### PR TITLE
BreinifySetup

### PR DESCRIPTION
**Pull request by akwong2**
### Change Logs
1. `isSetupComplete` to check if required info from client is gathered
2. Update rollup config to build axios call

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -42,6 +44,7 @@
 		"@testing-library/react": "^13.1.1",
 		"@types/jest": "^28.1.3",
 		"@types/react": "^18.0.0",
+		"axios": "^0.27.2",
```
Comment (timothyyue-b) [inline package.json:47]: I think `axios` would fall under `dependencies` instead of `devDependencies` since we are actually using it in our library code. I got the idea from this stackoverflow: [https://stackoverflow.com/questions/48288032/axios-dependencies-or-devdependencies](https://stackoverflow.com/questions/48288032/axios-dependencies-or-devdependencies). 

Let me know what you think

File: src/api/recommendations.ts [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+import axios from 'axios';
+
+// CONFIGS
+import { BreinifyGlobalConfigs } from '../setup';
+import { RECOMMENDATION_URL } from '../configs';
+
+// UTILS
+import { isSetupComplete } from '../utils';
+
+const INIT_ERROR = 'Please complete library initialization: BreinifySetup';
```
Comment (timothyyue-b) [inline src/api/recommendations.ts:11]: I think a better name here would be `INIT_ERROR_MESSAGE`

File: src/api/recommendations.ts [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+		secret: BreinifyGlobalConfigs.secret,
+		unixTimestamp: 1657152522,
+		recommendation: {},
+	};
+	return axios.post(RECOMMENDATION_URL, data).then((response) => {
+		const { data, status } = response;
+		if (status === 200) return data;
+		throw data;
+	});
+}
```
Comment (timothyyue-b) [inline src/api/recommendations.ts:27]: Is this just the initial version for testing?

File: src/configs/configs.ts [hunk 4]
```diff
@@ -0,0 +1,5 @@
+const API = 'https://api.breinify.com';
+
```
Comment (timothyyue-b) [inline src/configs/configs.ts:2]: I think it'll be a little more readable to have no newline between these constants, since they are kind of related

File: src/setup/setup.ts [hunk 5]
```diff
@@ -0,0 +1,17 @@
+type ConfigProps = {
+	apiKey?: string;
```
Comment (timothyyue-b) [inline src/setup/setup.ts:2]: I think the type here for `apiKey` is required right? We can also make use of Typescript's `Partial` or `Required` utility types to slightly tweak the base types. Let me know what you think.

File: tsconfig.json [hunk 6]
```diff
@@ -10,6 +10,7 @@
 		"strict": true,
 		"forceConsistentCasingInFileNames": true,
 		"esModuleInterop": true,
-		"baseUrl": "."
-	}
+		"baseUrl": "./",
+	},
+//	"include": ["src"]
```
Comment (timothyyue-b) [inline tsconfig.json:15]: Let's revert these changes, since it looks like its just commented code and the baseUrl is effectively the same

**Review by timothyyue-b — CHANGES_REQUESTED** [review]
Initial changes look good, I do have some inline questions / NIT comments before merging

File: package.json (hunk 1 above)
Comment (akwong2) [reply package.json:47]: yeah I agree, we had it in dep before but with all the messing around, I forgot to commit it

File: src/api/recommendations.ts (hunk 3 above)
Comment (akwong2) [reply src/api/recommendations.ts:27]: yeah just an initial version, wanted to go more in depth once I have your types

File: src/setup/setup.ts (hunk 5 above)
Comment (akwong2) [reply src/setup/setup.ts:2]: you're right, apiKey is the only one that's required. Everything else is optional. I could make a type for apiKey & secret individually to utilize `Partial` AND `Required`. Is that what you were thinking?

**Review by timothyyue-b — APPROVED** [review]
LGTM!
